File: frontend/src/types/followUp.ts
export interface FollowUp {
  id: number
  person_project_id: number
  content: string
  next_follow_time: string | null
  created_by: number
  created_by_name: string
  created_at: string
  updated_at: string
}

export interface FollowUpCreate {
  person_project_id: number
  content: string
  next_follow_time?: string | null
}

File: backend/app/schemas/follow_up.py
"""跟进记录 schemas。"""

from datetime import datetime

from pydantic import BaseModel

class FollowUpBase(BaseModel):
    """跟进记录基础模型。"""

    person_project_id: int
    content: str
    next_follow_time: datetime | None = None


class FollowUpCreate(FollowUpBase):
    """创建跟进记录。"""

    pass


class FollowUpResponse(FollowUpBase):
    """跟进记录响应。"""

    id: int
    created_by: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
